library/packages/src/lib/y2packager/product_reader.rb

```diff
@@ -120,6 +132,38 @@ def available_products
       products
     end
 
+    # read the available products, remove potential duplicates
+    # @return [Array<Hash>] pkg-bindings data structures
+    def available_products
+      # remove e.g. installed products
+      zypp_products.select { |p| p["status"] == :available || p["status"] == :selected }
+    end
+
+    # read the installed products
+    # @return [Array<Hash>] pkg-bindings data structures
+    def installed_products
+      # remove e.g. available or removed products
+      zypp_products.select { |p| p["status"] == :installed || p["status"] == :removed }
+    end
+
+    # find the installed base product
+    # @return[Hash,nil] the pkg-bindings product structure or nil if not found
+    def base_product
+      products = Yast::Pkg.ResolvableProperties("", :product, "")
+
+      # The base product is identified by the /etc/products.d/baseproduct symlink
+      # and because a symlink can point only to one file there can be only one base product.
+      # The "status" conditition is actually not required because that symlink is created
+      # only for installed products. (Just make sure it still works in case the libzypp
+      # internal implementation is changed.)
+      base = products.find do |p|
+        p["type"] == "base" && (p["status"] == :installed || p["status"] == :removed)
+      end
+
+      log.info("Found installed base product: #{base}")
+      base
```

Comment:
why not simple `installed_products.find { |p| p.type == "base" }` ?

Reply:
Because `Y2Packager::Product` does not have `type` attribute... :worried:

Reply:
but installed_products is from `zypp_products` which is `Pkg` products, not `Y2Packager::Product`

Reply:
I think you are hit by your own method names, because here is two installled_products, one in reader and second in Product class and both return different kind of object, which is a bit confusing.

Reply:
Oh, you are right, it's `Product.installed_products` vs. `ProductReader#installed_products`... :worried: